src/SourceLocator/Type/Composer/Factory/MakeLocatorForComposerJsonAndInstalledJson.php

```diff
@@ -38,19 +38,22 @@ public function __invoke(string $installationPath, Locator $astLocator): SourceL
             throw InvalidProjectDirectory::atPath($installationPath);
         }
 
-        $composerJsonPath  = $realInstallationPath . '/composer.json';
-        $installedJsonPath = $realInstallationPath . '/vendor/composer/installed.json';
+        $composerJsonPath = $realInstallationPath . '/composer.json';
 
         if (! file_exists($composerJsonPath)) {
             throw MissingComposerJson::inProjectPath($installationPath);
         }
 
+        /** @var array{autoload: array{classmap: array<int, string>, files: array<int, string>, psr-4: array<string, array<int, string>>, psr-0: array<string, array<int, string>>}, config: array{vendor-dir: array<string, array<int, string>>}}|null $composer */
+        $composer  = json_decode((string) file_get_contents($composerJsonPath), true);
+        $vendorDir = $composer['config']['vendor-dir'] ?? 'vendor';
```

Comment:
Does this throw any warnings if the keys don't exist, I can never remember sorry heh

Reply:
To my knowledge no. As the test don't throw any warning and it isn't set in the test, this should be safe.

Reply:
No, `??` is very much lax :-)